okapi-core/src/main/java/org/folio/okapi/managers/InternalModule.java

```diff
@@ -766,7 +766,16 @@ public static ModuleDescriptor moduleDescriptor(String okapiVersion) {
     TenantInstallOptions options = ModuleUtil.createTenantOptions(pc.getCtx().request());
     UUID installId = UUID.randomUUID();
     return tenantManager.installUpgradeCreate(tenantId, installId.toString(), pc, options, null)
-        .compose(res -> Future.succeededFuture(Json.encodePrettily(res)));
+        .compose(res -> {
+          String jsonResponse = Json.encodePrettily(res);
+          if (options.getAsync()) {
+            // using same location as install
+            String baseUri = pc.getCtx().request().uri().replace("/upgrade", "/install");
+            return location(pc, installId.toString(), baseUri, jsonResponse);
```

Comment:
Technically right. Maybe a little bit counter intuitive. Mentioning this in the guide should help.